automation/src/main/java/org/wso2/carbon/testgrid/automation/executers/TestNgExecuter.java

```diff
@@ -18,23 +18,65 @@
 
 package org.wso2.carbon.testgrid.automation.executers;
 
+import org.testng.TestNG;
 import org.wso2.carbon.testgrid.automation.exceptions.TestGridExecuteException;
 import org.wso2.carbon.testgrid.automation.executers.common.TestExecuter;
 import org.wso2.carbon.testgrid.common.Deployment;
 
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+
 /**
  * This class is responsible for Executing TestNG tests.
  */
 public class TestNgExecuter implements TestExecuter {
+    private String testGridFolder;
+    private String testName;
 
     @Override
-    public void execute(String script, Deployment deployment) {
+    public void execute(String jarFilePath, Deployment deployment) {
+        File jarFile = new File(jarFilePath);
+        String jarName = jarFile.getName().substring(0, jarFile.getName().lastIndexOf("."));
+
+        loadJarToClasspath(jarFile);
 
+        TestNG testng = new TestNG();
+        testng.setTestJar(jarFilePath);
+        testng.setOutputDirectory(testGridFolder + "/Results/TestNG/" + jarName);
+        testng.run();
     }
 
     @Override
     public void init(String testGridFolder, String testName) throws TestGridExecuteException {
+        this.testGridFolder = testGridFolder;
+        this.testName = testName;
+    }
+
+    public static void loadJarToClasspath(File jarFile) {
+        if (jarFile.isFile()) {
+            URL url = null;
+            try {
+                url = jarFile.toURL();
+                URL[] urls = new URL[] { url };
+                ClassLoader cl = new URLClassLoader(urls);
+                Thread.currentThread().setContextClassLoader(cl);
+            } catch (MalformedURLException e) {
+                e.printStackTrace();
```

Comment:
We should introduce a logging framework. How about SLF4J?

Reply:
Yes we should. This will be treated as a new feature. Created issue #52